packages/frontend/src/providers/DashboardProvider.tsx

```diff
@@ -266,9 +281,21 @@ export const DashboardProvider: React.FC<{
         }
     }, [dashboard?.filters, overridesForSavedDashboardFilters]);
 
-    // Gets filters from URL and storage after redirect
+    // Gets filters and dateZoom from URL and storage after redirect
     useMount(() => {
         const searchParams = new URLSearchParams(search);
+
+        // Date zoom
+        const dateZoomParam = searchParams.get('dateZoom');
+        if (dateZoomParam) {
+            const dateZoom = Object.values(DateGranularity).find(
+                (granularity) =>
+                    granularity.toLowerCase() === dateZoomParam?.toLowerCase(),
+            );
+            if (dateZoom) setDateZoomGranularity(dateZoom);
```

Comment:
> Also, can we remove the dateZoom param if !dateZoom? I think it would be cool to remove unwanted dateZooms

This is done here right ? https://github.com/lightdash/lightdash/pull/8151/files#diff-75153232c9b14b6bd094c7f14226841cdaaec45925707d3819bbde85585b440aR163

Reply:
You can't use a string to get an enum 

![image](https://github.com/lightdash/lightdash/assets/1983672/2ebf0494-0b35-4d8d-9d72-afc5b8254ff2)

Afaik the only way to do something similar is to use casting like

`DateGranularity[dateZoomParam as keyof typeof DateGranularity];`

Reply:
> Afaik the only way to do something similar is to use casting like

ah, that's annoying, ignore!

> This is done here right ? https://github.com/lightdash/lightdash/pull/8151/files#diff-75153232c9b14b6bd094c7f14226841cdaaec45925707d3819bbde85585b440aR163

If you visit here:
https://lightdash-pr-8151.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/e0e8ab5d-e950-4591-9386-d9a22658162d/view?dateZoom=testing

The `dateZoom` param doesn't get removed when it's invalid. But don't want to block this, so all good!